apps/rush-lib/src/logic/Git.ts

```diff
@@ -133,11 +134,15 @@ export class Git {
   /**
    * Get the folder where Git hooks should go for the current working tree.
    * Returns undefined if the current path is not under a Git working tree.
+   * Respects `git config core.hooksPath`.
    */
   public getHooksFolder(): string | undefined {
     const repoInfo: gitInfo.GitRepoInfo | undefined = this.getGitInfo();
-    if (repoInfo && repoInfo.worktreeGitDir) {
-      return path.join(repoInfo.worktreeGitDir, 'hooks');
+    const hooksResult: IResultOrError<string> = this._tryGetGitHooksPath();
+    const hooksPath: string = hooksResult.result ?? '.git/hooks';
+
+    if (repoInfo?.root) {
+      return path.resolve(repoInfo.root, hooksPath);
     }
```

Comment:
AFAIK git can be complicated and I don't know anything but the most basic usage, so I'm not sure this covers 100% of possibilities? Should be no worse than how it was before, I guess

Reply:
Apparently there's quite a bit more involved in finding the `worktreeGitDir` vs. just finding the `root`. See: https://github.com/rwjblue/git-repo-info/blob/d3ab418ef8b392eabbe911a37871708b15201b70/index.js#L13

So it seems it would be safer to default to `worktreeGitDir` instead of doing a `path.join(...)` with the `.git` folder being expected at the root.

Reply:
So, I need some help here. 

In your average repo (like the `rushstack`) `git rev-parse` returns me the hook directory relative to the root, not the `.git`. Since `rev-parse` is doing it relative to the directory where it's run (i.e. if I run it manually 5 folders deep in the repo it returns `../../../../../.git/hooks` for unset param).

I assume this just means that `rush` always runs `git` commands at the repo root. If there is a variable for "Directory where we run git" I'd rather use that, but I haven't found anything like it.

This would mean that It's safe to leave this as is even in the worktree case. However, it would be great if someone with the actual setup tested it. If there's no one and it has to be me, it'll take me some time to understand what those things are and how to set them up.

Another thing that can be done is use [-path-format=absolute](https://git-scm.com/docs/git-rev-parse#Documentation/git-rev-parse.txt---path-formatabsoluterelative) on `rev-parse`, and not guess anything, BUT it's a feature from Q1 2021, and wasn't available on my machine, so I decided that it's too early to depend on it.

Relevant reading:
The git docs on the [param itself](https://git-scm.com/docs/git-config#Documentation/git-config.txt-corehooksPath) are quite laconic.
The docs on [rev-parse](https://git-scm.com/docs/git-rev-parse#Documentation/git-rev-parse.txt---git-pathltpathgt) are barely coherent if you ask me (or maybe just require deeper git knowledge).

___
P.S. - It's `path.resolve`, not `.join` - `hooksPath` can be absolute instead, and `resolve` will respect that unlike `join`

Reply:
@Faithfinder Agree that depending on `-path-format=absolute` isn't the right solution here.

As for the first part, I agree with your logic here. I'm not too personally familiar with the reasoning behind why we used `worktreeGitDir`, but looking into the docs, it may not even be the correct value to use here. The git docs you provided state that the `core.hooksPath` should be postfixed to `GIT_DIR` to get the absolute hooks path, which implies it should be located in the main worktree, so really we shouldn't need to worry about the worktree at all.

We run the git commands in this class with a working dir of `this._rushConfiguration.rushJsonFolder`. So, maybe we should have 2 cases here where:
- `path.resolve(this._rushConfiguration.rushJsonFolder, hooksPath)` is used in the case where we have the `hooksPath`, since it's returned relative to the working directory
- `path.resolve(repoInfo.root, '.git', 'hooks')` is used in the case where we don't

It's extremely rare to see a repo with a Rush configuration somewhere other than the root of the repo (and it would likely cause other issues... but that's beside the point), but it is possible. This feels like the _most correct_ way of doing it as far as I can tell... Thoughts?

Reply:
Yeah, sounds reasonable. I've made the changes and pushed them

Reply:
I recently made changes to a family of git commands to, in fact, always run them from the git root, specifically to support having rush.json in a subfolder. I use this function here to obtain said root for comparison purposes: https://github.com/microsoft/rushstack/blob/fa7f808d23e1f77047fffed56ef9390168c8111f/libraries/package-deps-hash/src/getRepoState.ts#L138